### DISCUSSION TITLE
What are the best practices for folder organization with Edge Functions and Cloudflare Workers?

### DISCUSSION THREAD

**Discussion by EliaTolin** [supabase/supabase#30047] [Questions]
Hi!
I use Supabase Edge Functions and Cloudflare Workers, and I want to find the best practices for organizing my project.
This is my current folder structure for the project:
```
project_backend/
├── supabase/
│   └── functions/
│       └── _shared/
│           ├── my_beautiful_shared_code1/
│           │   └── shared_code1
│           ├── my_beautiful_shared_code2
│           └── my_beautiful_shared_codeN
└── workers/
    ├── _shared/
    │   └── my_beautiful_code_only_for_workers
    ├── compute-foo-worker/
    │   ├── .wrangler
    │   ├── node_modules
    │   ├── ...
    │   ├── src/
    │   │   └── index.ts
    │   ├── ...
    │   ├── wrangler.toml
    │   ├── tsconfig.json
    │   └── package.json
    ├── compute-alice-worker/
    │   └── ...
    └── compute-bob-worker/
        └── ...
```

For example, in my `index.ts `inside` compute-foo-worker`, I want to use `shared_code1` from `my_beautiful_shared_code1`, but `shared_code1` uses a package, for example, `Stripe`.

`shared_code1` is used both by Supabase Edge Functions and Cloudflare Workers.

Right now, the organization is unclear to me.
What are the best practices for handling this?

**Comment by fbatiga** [comment]
what are you using the cloudfare workers for ?

**Reply by EliaTolin** [reply]
batches of many Long tasks (2/3 minutes)

**Comment by EliaTolin** [comment]
Any suggestion?
